.github/workflows/scripts/bump-go.sh

```diff
@@ -0,0 +1,135 @@
+set -euo pipefail
+
+usage() {
+  echo "Usage: $0 [--apply|-a] <path/to/go.mod>" >&2
+  exit 1
+}
+
+require_cmd() {
+  command -v "$1" >/dev/null 2>&1 || {
+    echo "Error: '$1' is required but was not found in PATH" >&2
+    exit 1
+  }
+}
+
+# ---- Preconditions ----------------------------------------------------------
+for cmd in curl jq git gh; do
+  require_cmd "$cmd"
+done
+
+# ---- Argument parsing -------------------------------------------------------
+APPLY=0
+GO_MOD=""
+
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    --apply|-a) APPLY=1 ;;
+    -h|--help)  usage ;;
+    *)          [[ -z "$GO_MOD" ]] && GO_MOD="$1" || usage ;;
+  esac
+  shift
+done
+
+[[ -z "$GO_MOD" ]] && usage
+[[ -f "$GO_MOD" ]] || { echo "Error: '$GO_MOD' not found" >&2; exit 1; }
+
+# ---- Discover latest stable Go release --------------------------------------
+echo "Fetching latest stable Go version…"
+LATEST_JSON=$(curl -fsSL https://go.dev/dl/?mode=json | jq -c '[.[] | select(.stable==true)][0]')
+FULL_VERSION=$(jq -r '.version' <<< "$LATEST_JSON")        # e.g. go1.23.4
+TOOLCHAIN_VERSION="${FULL_VERSION#go}"                     # e.g. 1.23.4
+GO_DIRECTIVE_VERSION=$(cut -d. -f1-2 <<< "$TOOLCHAIN_VERSION")
+
+echo "  → go : $GO_DIRECTIVE_VERSION"
+echo "  → toolchain : $TOOLCHAIN_VERSION"
+
+# ---- Prepare Git branch ---------------------------------------------------
+CURRENT_GO_DIRECTIVE=$(grep -E '^go ' "$GO_MOD" | awk '{print $2}')
+CURRENT_TOOLCHAIN_DIRECTIVE=$(grep -E '^toolchain ' "$GO_MOD" | awk '{print $2}' | cut -d' ' -f2)
+
+if [[ "$CURRENT_GO_DIRECTIVE" = "$GO_DIRECTIVE_VERSION" && \
+      "$CURRENT_TOOLCHAIN_DIRECTIVE" = "go$TOOLCHAIN_VERSION" ]]; then
+  echo "Already on latest Go version: $CURRENT_GO_DIRECTIVE (toolchain: $CURRENT_TOOLCHAIN_DIRECTIVE)"
+  exit 0
+fi
+
+BRANCH="bump-go-$TOOLCHAIN_VERSION"
+cleanup() {
+  git checkout - >/dev/null 2>&1 || true
+  git branch -D "$BRANCH" >/dev/null 2>&1 || true
+}
+trap cleanup EXIT
+
+echo "Creating branch $BRANCH"
+git switch -c "$BRANCH" >/dev/null 2>&1
+
+# ---- Patch go.mod -----------------------------------------------------------
+if [[ "$CURRENT_GO_DIRECTIVE" != "$GO_DIRECTIVE_VERSION" ]]; then
+  sed -Ei.bak "s/^go [0-9]+\.[0-9]+.*$/go $GO_DIRECTIVE_VERSION/" "$GO_MOD"
+  echo "  • go directive $CURRENT_GO_DIRECTIVE → $GO_DIRECTIVE_VERSION"
+fi
+
+if [[ "$CURRENT_TOOLCHAIN_DIRECTIVE" != "go$TOOLCHAIN_VERSION" ]]; then
+  sed -Ei.bak "s/^toolchain go[0-9]+\.[0-9]+\.[0-9]+.*$/toolchain go$TOOLCHAIN_VERSION/" "$GO_MOD"
+  echo "  • toolchain $CURRENT_TOOLCHAIN_DIRECTIVE → go$TOOLCHAIN_VERSION"
+fi
+
+rm -f "$GO_MOD.bak"
+
+git add "$GO_MOD"
+
+# ---- Commit -----------------------------------------------------------------
+COMMIT_MSG="Bump Go to $TOOLCHAIN_VERSION"
+git commit -m "$COMMIT_MSG" >/dev/null
+COMMIT_HASH=$(git rev-parse --short HEAD)
+
+PR_TITLE="$COMMIT_MSG"
+
+# ---- Check for existing PR ------------------------------------------
```

Comment:
My OCD brain says: "either keep these dashed lines lengths the same, or get rid of it.", but that's just my brain, feel free to ignore. :D